controllers/replicationdestination.go

```diff
@@ -0,0 +1,92 @@
+package controllers
+
+import (
+	"fmt"
+
+	volsyncv1alpha1 "github.com/backube/volsync/api/v1alpha1"
+	"github.com/go-logr/logr"
+	pvcv1alpha1 "github.com/konveyor/volume-snapshot-mover/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *DataMoverRestoreReconciler) CreateReplicationDestination(log logr.Logger) (bool, error) {
+
+	// get datamoverrestore from cluster
+	dmr := pvcv1alpha1.DataMoverRestore{}
+	if err := r.Get(r.Context, r.NamespacedName, &dmr); err != nil {
+		r.Log.Error(err, "unable to fetch DataMoverRestore CR")
+		return false, err
+	}
+
+	// get datamoverbackup from cluster
+	// TODO: name
+	dmb := pvcv1alpha1.DataMoverBackup{}
+	if err := r.Get(r.Context, types.NamespacedName{Name: "datamoverbackup-sample", Namespace: r.NamespacedName.Namespace}, &dmb); err != nil {
+		r.Log.Error(err, "unable to fetch DataMoverBackup CR")
+		return false, err
+	}
+
+	// define replicationDestination to be created
+	repDestination := &volsyncv1alpha1.ReplicationDestination{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-rep-dest", dmr.Name),
+			Namespace: r.NamespacedName.Namespace,
+		},
+	}
+
+	// Create ReplicationDestination in OADP namespace
+	op, err := controllerutil.CreateOrUpdate(r.Context, r.Client, repDestination, func() error {
+
+		return r.buildReplicationDestination(repDestination, &dmb, &dmr)
+	})
+	if err != nil {
+		return false, err
+	}
+
+	if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated {
+		r.EventRecorder.Event(repDestination,
+			corev1.EventTypeNormal,
+			"ReplicationDestinationReconciled",
+			fmt.Sprintf("%s replicationdestination %s", op, repDestination.Name),
+		)
+	}
+	return true, nil
+}
+
+func (r *DataMoverRestoreReconciler) buildReplicationDestination(replicationDestination *volsyncv1alpha1.ReplicationDestination, dmb *pvcv1alpha1.DataMoverBackup, dmr *pvcv1alpha1.DataMoverRestore) error {
+
+	// get restic secret created by controller
+	resticSecretName := fmt.Sprintf("%s-secret", dmr.Name)
+	resticSecret := corev1.Secret{}
+	if err := r.Get(r.Context, types.NamespacedName{Namespace: r.NamespacedName.Namespace, Name: resticSecretName}, &resticSecret); err != nil {
+		r.Log.Error(err, "unable to fetch Restic Secret")
+		return err
+	}
+
+	stringCapacity := dmb.Status.SourcePVCData.Size
```

Comment:
Should we populate this field from dmr instead of dmb?

Reply:
Would it be a good idea to put this on DMR?

Reply:
My understanding was DMB CR will not be restored. So when we create DMR CR, we will add all the fields from DMB that we need. 
Would DMR controller know the correct DMB CR to pick?

Reply:
I gotcha. I guess this is something we will need to hash out once we are able to test getting DMB from backup? In the meantime, okay to leave as is?

Reply:
Yes. May be add a comment/todo so that we dont forget and its good to merge :) Thanks Emily :)